src/page-multi-facility/ReSyncRefFacilityModal.tsx

```diff
@@ -0,0 +1,266 @@
+        <Spacer />
+        Type of System: {systemType}
+      </TitleText>
+    )}
+  </SyncDataTitleContainer>
+);
+
+const FacilitiesSelect: React.FC<{
+  value: Facility["id"] | undefined;
+  selections: { [key in ReferenceFacility["id"]]: Facility["id"] };
+  facilities: Facilities;
+  onChange: (value: Facility["id"] | undefined) => void;
+}> = ({ selections, value, facilities, onChange }) => {
+  const disabledOption = (facilityId: Facility["id"]) =>
+    Object.values(selections).includes(facilityId);
+  return (
+    <FacilitySelectContainer>
+      <InputSelect
+        onChange={(event) => {
+          console.log("onchange event", event.target.value);
+          const value = event.target.value;
+          onChange(value === "" ? undefined : value);
+        }}
+        value={value || ""}
+      >
+        <option value={""}>Select a facility</option>
+        {facilities.map((facility: Facility) => {
+          return (
+            <option
+              key={facility.id}
+              value={facility.id}
+              disabled={disabledOption(facility.id)}
+            >
+              {facility.name}
+            </option>
+          );
+        })}
+      </InputSelect>
+    </FacilitySelectContainer>
+  );
+};
+
+const filterSyncedFacilities = (
+  syncedRefFacilities: FacilityReferenceMapping,
+) => {
+  return (facility: Facility) => {
+    return !Object.keys(syncedRefFacilities).includes(facility.id);
+  };
+};
+
+interface ModalProps {
+  open: boolean;
+  stateName: ModelInputs["stateName"];
+  systemType: Facility["systemType"];
+  onClose: () => void;
+}
+
+const ReSyncRefFacilityModal: React.FC<ModalProps> = ({
+  open,
+  stateName,
+  systemType,
+  onClose,
+}) => {
+  const rejectionToast = useRejectionToast();
+  const [scenarioState, dispatchScenarioUpdate] = useScenario();
+  const {
+    state: { facilities: facilitiesState, referenceFacilities },
+  } = useFacilities();
+
+  const [selections, setSelections] = useState<{
+    [refFacilityId: string]: Facility["id"];
+  }>({});
+
+  const scenario = scenarioState.data;
+  const syncedRefFacilities = scenario?.[referenceFacilitiesProp] || {};
+
+  if (!open) return null;
+
+  const facilities = Object.values(facilitiesState).filter(
+    filterSyncedFacilities(syncedRefFacilities),
+  );
+
+  async function handleSave() {
+    if (Object.keys(selections)) {
+      const facilityIdToReferenceId = invert(selections);
+
+      await rejectionToast(
+        saveScenario({
+          ...scenario,
+          [referenceFacilitiesProp]: Object.assign(
+            {},
+            scenario?.[referenceFacilitiesProp],
+            facilityIdToReferenceId,
+          ),
+        }).then((savedScenario) => {
+          if (savedScenario) dispatchScenarioUpdate(savedScenario);
+        }),
+      );
+    }
+    onClose();
+  }
+
+  return (
+    <ModalDialog
+      open={open}
+      title={<SyncDataTitle stateName={stateName} systemType={systemType} />}
```

Comment:
`ModalDialog` takes in a `width` property which you'll probably want to use to reduce the size of the modal from its default.